lib/ecto/query/builder.ex

```diff
@@ -536,14 +553,47 @@ defmodule Ecto.Query.Builder do
   """
   def keyword!(kw) do
     unless Keyword.keyword?(kw) do
-      raise ArgumentError, "to prevent sql injection, only a keyword list may be interpolated " <>
-                           "as the first argument to `fragment/1` with the `^` operator, got `#{inspect kw}`"
+      message = """
+      to prevent sql injection, only a keyword list may be interpolated
+      as the first argument to `fragment/1` with the `^` operator, got `#{inspect kw}`
+      """
+      message = if is_binary(kw) do
+        message <> """
+
+        For interpolated strings use `unsafe_fragment/1` to pass the argument
```

Comment:
I think "interpolated strings" is ambiguous because of string interpolation. I think we can have a single message like this:

```
"""
to prevent SQL injection attacks, fragment allows only keyword lists to be
interpolated as the first argument via the `^` operator.

If you are in a rare situation where the string that must be sent to the database
is defined dynamically, you can use `unsafe_fragment/1` to bypass Ecto checks
and send the string as is. Use `unsafe_fragment/1` only as last resort and wisely.

Got: `#{inspect kw}`
"""
```